src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1108,6 +1124,18 @@ private static ObjectType GetParseCidrReturnType()
             }, null);
         }
 
+        private static ObjectType GetParseUriReturnType()
+        {
+            return new ObjectType("parseUri", TypeSymbolValidationFlags.Default, new[]
+            {
+                new TypeProperty("scheme", LanguageConstants.String),
```

Comment:
If we also end up using this type as the type of the input object into `buildUri()`, we will need to add `TypePropertyFlags.Required` to the `scheme` and `host` properties.